utils.lisp

```diff
@@ -82,3 +82,53 @@
 
 (define-condition user-error (simple-error) ())
 
+(defun singlep (l)
+  (= 1 (length l)))
+
+(defmacro comment (&rest body)
+  (declare (ignore body))
+  nil)
+
+;; The original idea for this was that I wanted
+;; a way to link using a minimal unique prefix.
+;;
+;; If my section heading is something long,
+;; I don't want to have to type out the entire
+;; Click [here](@Some long section heading that is easy to typo) to read more!
+;;
+;; So I was going to use a trie to lookup the Chapter/Section ID given
+;; a title prefix. If it ends up being unique, we're good to go. If it's
+;; not unique, we can pick one and leave a warning.
+;;
+;; I haven't yet gotten close to finishing that implementation. So it
+;; doesn't make much sense to have this data structure right now.
+;; But oh well. Whatever. It works for an MVP of the chapter/section linking stuff.
```

Comment:
^ Of course that could also be done with simple string matching. I was also thinking this could be useful for something like letting you use `[some section](@some section)` if it's unique but `[some section](@some chapter:@some-section)` if the section title wasn't unique. (Etc. for `@some file name`). Basically... I had a very loosely woven idea that it might be useful and it seemed like a fun way to practice LISP. I don't think it's actually worthwhile for anything.

Reply:
I had not thought of this idea. I am perfectly open to giving it a try and think its a cool idea. I am also just fine with forcing them to write out the title.